assets/app/main.js

```diff
@@ -0,0 +1,38 @@
+
+import ReactDOM from 'react-dom';
+import React from 'react';
+
+import buildActions from './actions/buildActions';
+import siteActions from './actions/siteActions';
+import userActions from './actions/userActions';
+
+import App from './components/app';
+import Header from './components/header';
+
+import router from './router';
+import store from './store';
+
+const mainEl = document.querySelector('#js-app');
+
+store.subscribe(() => {
+  const state = store.getState();
+  const isLoggedIn = (state.user) ? true : false;
+  if (!isLoggedIn) return;
+
+  ReactDOM.render(
```

Comment:
Yes. Well, we're having each component call `.render()` and the cumulative result is diffed against the current DOM from the mount node.

My understanding was that this was an OK method because the DOM won't actually change if there are no updates to make.

Is there a better way to do this?